library/src/main/kotlin/com/pusher/platform/schedulers.kt

```diff
@@ -1,41 +1,87 @@
 package com.pusher.platform
 
 import android.os.Handler
-import android.os.HandlerThread
 import android.os.Looper
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
 
 typealias Action = () -> Unit
 
-sealed class AndroidScheduler(private val handler: Handler) : Scheduler {
+class BackgroundScheduler(
+    threadLimit: Int = 50,
+    theadAliveTime: Long = 60
```

Comment:
What does this control?

Reply:
> when the number of threads is greater than the core, this is the maximum time that excess idle threads will wait for new tasks before terminating.

Maybe we could put a lower value for this one, instead of 60 seconds

Reply:
Ah okay. I'll leave that up to you. Seems fine to me though!